src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/NuGetDependenciesSubTreeProvider.cs

```diff
@@ -216,96 +227,259 @@ public override IDependencyNode GetDependencyNode(DependencyNodeId id)
+            var resolvedChanges = ResolvedReferenceRuleNames.Where(x => changes.Keys.Contains(x))
+                                                            .Select(ruleName => changes[ruleName])
+                                                            .ToImmutableHashSet();
+            var dependenciesChange = new NugetDependenciesChange();
+            var newDependencies = new HashSet<DependencyMetadata>();
+            foreach (var change in resolvedChanges)
+            {
+                if (!change.Difference.AnyChanges)
+                {
+                    continue;
+                }
+
+                foreach (string removedItemSpec in change.Difference.RemovedItems)
+                {
+                    var metadata = CurrentSnapshot.RemoveDependency(removedItemSpec);
+                    var itemNode = rootTreeNodes.FirstOrDefault(
+                                    x => x.Id.ItemSpec.Equals(removedItemSpec, StringComparison.OrdinalIgnoreCase));
+                    if (itemNode != null)
+                    {
+                        dependenciesChange.RemovedNodes.Add(metadata);
+                    }
+                }
+
+                foreach (string changedItemSpec in change.Difference.ChangedItems)
+                {
+                    var properties = GetProjectItemProperties(change.After, changedItemSpec);
+                    if (properties == null)
+                    {
+                        continue;
+                    }
+
+                    var metadata = CurrentSnapshot.UpdateDependency(changedItemSpec, properties);
+                    var itemNode = rootTreeNodes.FirstOrDefault(
+                                    x => x.Id.ItemSpec.Equals(changedItemSpec, StringComparison.OrdinalIgnoreCase));
+                    if (itemNode != null)
+                    {
+                        dependenciesChange.UpdatedNodes.Add(metadata);
+                    }
+                }
+
+                foreach (string addedItemSpec in change.Difference.AddedItems)
+                {
+                    var properties = GetProjectItemProperties(change.After, addedItemSpec);
+                    if (properties == null)
+                    {
+                        continue;
+                    }
+
+                    var newDependency = CurrentSnapshot.AddDependency(addedItemSpec, properties);
+                    newDependencies.Add(newDependency);
+                }
+            }
+
+            // Note: currently deisgn time build is limited and is not aware of conditional on TFM 
+            // PackageReference items: Unresolved PackageReference items for conditional TFMs are not sent.
+            // Thus we will display conditional PackageReferences if they were resolved and are in assets.json.
+            // This limitation should go away, when we have final design for cross target dependencies and 
+            // DesignTime build.
```

Comment:
however "ResolvedPackageReferences" are sent accross all TFMs since they are sent after reading assets json file which has info for all targets.

---
In reply to: [85849338](https://github.com/dotnet/roslyn-project-system/pull/675#discussion_r85849338) [](ancestors = 85849338,85848189)